include/picongpu/plugins/hdf5/WriteSpecies.hpp

```diff
@@ -62,6 +62,58 @@ TYPE_ARRAY(UInt64_5, H5T_INTEL_U64, uint64_t, 5);
 
 using namespace splash;
 
+namespace detail
+{
+    template<
```

Comment:
Note this can be also written as:

```C++
template<
    typename T_FrameType
>
struct GetChargeOrZero
{
    static constexpr bool hasChargeRatio = pmacc::traits::HasFlag<
        T_FrameType,
         chargeRatio<>
     >::type::value;

     auto operator()()
     -> typename std::enable_if< hasChargeRatio, float_X >::type
     {
         return frame::getCharge< T_FrameType >();
     }

     auto operator()()
     -> typename std::enable_if< !hasChargeRatio, float_X >::type
     {
         return float_X( 0. );
     }
};
```

Reply:
A little bit docu: https://stackoverflow.com/questions/14600201/why-should-i-avoid-stdenable-if-in-function-signatures